src/module_manager.ts

```diff
@@ -0,0 +1,88 @@
+import { join } from 'path'
+import { readdirSync } from 'fs'
+
+import { TelegramClient } from 'telegram'
+import { NewMessageEvent } from 'telegram/events'
+
+import { Module, isModule } from './module'
+
+export class ModuleManager {
+	private modules = new Map<string, [Module, boolean]>()
+
+	constructor(
+		private client: TelegramClient,
+		public disabled = new Set<string>()
+	) {}
+
+	handler = async (event: NewMessageEvent) => {
+		for (const [, [{ name, handlers }, disableable]] of this.modules) {
+			if (disableable && this.disabled.has(name)) {
+				return
+			}
+			for (const handler of handlers) {
+				if (await handler.check(this.client, event)) {
+					try {
+						await handler.handle(this.client, event)
+					} catch (err) {
+						console.error(err)
+					}
+				}
+			}
+		}
+	}
+
+	install(module: Module, disableable = false) {
+		if (this.modules.has(module.name)) {
+			throw new Error(`Module ${module.name} is already installed`)
+		}
+		this.modules.set(module.name, [module, disableable])
+	}
+
+	installMultiple(disableable: boolean, ...modules: Module[]) {
+		for (const module of modules) {
+			this.install(module, disableable)
+		}
+	}
```

Comment:
It can be better in terms of readability if we keep the arguments order similar in `install` and `installMultiple`

```diff
--installMultiple(disableable: boolean, ...modules: Module[]) {
++installMultiple(modules: Module[], disableable: boolean) {
```
or vice-versa